daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -229,7 +230,7 @@ private[lf] object Speedy {
     private[speedy] def getEnvFree(i: Int): SValue = frame(i)
 
     @inline def pushEnv(v: SValue): Unit = {
-      env.add(v)
+      discard[Boolean](env.add(v))
```

Comment:
Why do we need this `[Boolean]` ?

Reply:
you don’t need it but it can be nice to be explicit in what you are discarding both for readability and to ensure that if the type changes it blows up and you need to reconsider whether dropping is still the right thing to do.
I have no strong feelings on whether it makes sense in this particular instance or not, I think it’s a case by case decison, sometimes it helps sometimes it doesn’t.

Reply:
:+1:

Reply:
I couldn't imagine what was being returned by `ArrayList.add`, so I checked some docs:
https://www.tutorialspoint.com/java/util/arraylist_add.htm
And discovered:
```
Return Value
This method returns true.
```
which made me so cross I thought to document with the type.

Reply:
I wonder if there is an easy way to extend the `ArrayList` class so the `add` method returns `Unit`, instead of having to `discard` at every point :shrug: